pkg/network/tracer/connection/failure/matching.go

```diff
@@ -104,22 +108,24 @@ func (fc *FailedConns) MatchFailedConn(conn *network.ConnectionStats) {
 	if fc == nil {
 		return
 	}
+
+	fc.Lock()
```

Comment:
Probably best to do this after the connection type check, so you don't acquire the lock unnecessarily.